itest/lnd_remote_signer_test.go

```diff
@@ -416,13 +737,119 @@ func testRemoteSignerTaproot(ht *lntest.HarnessTest) {
 			}
 		},
 	}
+}
 
-	_, watchOnly, carol := prepareRemoteSignerTest(ht, tc)
-	tc.fn(ht, watchOnly, carol)
+func testRemoteSignerTaproot(ht *lntest.HarnessTest) {
+	executeRemoteSignerTestCase(ht, taprootTestCase(false), false)
+}
+
+func testRemoteSignerTaprootOutbound(ht *lntest.HarnessTest) {
+	executeRemoteSignerTestCase(ht, taprootTestCase(true), true)
+}
+
+// testOutboundRSMacaroonEnforcement tests that a valid macaroon including
+// the `remotesigner` entity is required to connect to a watch-only node that
+// uses an outbound remote signer, while the watch-only node is in the state
+// where it waits for the signer to connect.
+func testOutboundRSMacaroonEnforcement(ht *lntest.HarnessTest) {
+	// Ensure that the watch-only node uses a configuration that requires an
+	// outbound remote signer during startup.
+	watchOnlyArgs := []string{
+		"--remotesigner.enable",
+		"--remotesigner.allowinboundconnection",
+		"--remotesigner.timeout=15s",
+		"--remotesigner.requesttimeout=15s",
+	}
+
+	// Create the watch-only node. Note that we require authentication for
+	// the watch-only node, as we want to test that the macaroon enforcement
+	// works as expected.
+	watchOnly := ht.CreateNewNode("WatchOnly", watchOnlyArgs, nil, false)
+
+	startChan := make(chan error)
+
+	// Start the watch-only node in a goroutine as it requires a remote
+	// signer to connect before it can fully start.
+	go func() {
+		startChan <- watchOnly.Start(ht.Context())
+	}()
+
+	// Wait and ensure that the watch-only node reaches the state where
+	// it waits for the remote signer to connect, as this is the state where
+	// we want to test the macaroon enforcement.
+	err := wait.Predicate(func() bool {
```

Comment:
nit: We almost never use it this now, as `wait.NoError` gives us more verbose errors - will find time and create a PR to remove that method.